pkg/capture/provider/network_capture_unix.go

```diff
@@ -361,16 +364,44 @@ func (ncp *NetworkCaptureProvider) Cleanup() error {
 	return nil
 }
 
-func obtainIptablesMode() string {
+type iptablesMode string
+
+const (
+	legacyIptablesMode iptablesMode = "legacy"
+	nftIptablesMode    iptablesMode = "nft"
+)
+
+func obtainIptablesMode(l *log.ZapLogger) iptablesMode {
 	// Since iptables v1.8, nf_tables are introduced as an improvement of legacy iptables, but provides the same user
 	// interface as legacy iptables through iptables-nft command.
 	// based on: https://github.com/kubernetes-sigs/iptables-wrappers/blob/97b01f43a8e8db07840fc4b95e833a37c0d36b12/iptables-wrapper-installer.sh
-	legacySaveOut, _ := exec.Command("iptables-legacy-save").CombinedOutput()
+
+	// when both iptables modes available, we choose the one with more rules.
+	nftIptablesModeAvaiable := true
+	legacyIptablesModeAvaiable := true
+	legacySaveOut, err := exec.Command("iptables-legacy-save").CombinedOutput()
+	if err != nil && strings.Contains(err.Error(), "command not found") {
```

Comment:
Grepping the error string is an anti-pattern. To accomplish this, you're better off using `exec.LookPath("iptables-legacy-save")`. This looks up the executable in the same way and expands it to an absolute path. It returns an error if the executable was not found. If you still need to run `iptables-legacy-save`, you can pass that absolute path to `exec.Command` and just handle any unexpected errors without trying to interpret them.

Reply:
Hey @timraymond, I realized using grep is not good also considering the "not found" error message varies in multiple platforms. So the latest change is just to decide the right mode from the output, normally the right mode should gives more results/rules than the unexpected one.